src/core.c/List.pm6

```diff
@@ -190,6 +190,21 @@ my class List does Iterable does Positional { # declared in BOOTSTRAP
         nqp::p6bindattrinvres(result,List,'$!todo',todo)
     }
 
+    method over-iterator(List:D: Iterator $iter --> List:D) {
+        # from-iterator only copies type-level information; here we copy
+        # everything bar the elements (e.g. container descriptors, shapes)
+        nqp::stmts(
+          (my \buffer := nqp::clone(self)),
```

Comment:
I find the use of the name `buffer` confusing here, as it is in fact a `List` object.  In the sibling methods `buffer` is always an `IterationBuffer`.

Reply:
Right, it's `result` elsewhere?

Reply:
Looks like  :-)